src/uploader.ts

```diff
@@ -26,48 +54,48 @@ export default class Uploader {
    *
    * @param {Function} onPreview - callback fired when preview is ready
    */
-  uploadSelectedFile({ onPreview }) {
-    const preparePreview = function (file) {
+  uploadSelectedFile({ onPreview }: UploadOptions) {
+    const preparePreview = function (file: File) {
       const reader = new FileReader();
 
       reader.readAsDataURL(file);
       reader.onload = (e) => {
-        onPreview(e.target.result);
+        onPreview((e.target as FileReader).result as string);
       };
     };
 
     /**
      * Custom uploading
      * or default uploading
      */
-    let upload;
+    let upload: Promise<UploadResponseFormat>;
 
     // custom uploading
     if (this.config.uploader && typeof this.config.uploader.uploadByFile === 'function') {
-      upload = ajax.selectFiles({ accept: this.config.types }).then((files) => {
+      upload = ajax.selectFiles({ accept: this.config.types || ''}).then((files: File[]) => {
         preparePreview(files[0]);
 
-        const customUpload = this.config.uploader.uploadByFile(files[0]);
+        const customUpload = this.config.uploader && this.config.uploader.uploadByFile && this.config.uploader.uploadByFile(files[0]);
```

Comment:
seems like it is already checked above, so you don't need to check for `uploader` and `uploadByFile` emptiness

Reply:
in building phase, i get this error if it is not checked again:
```
src/uploader.ts:77:30 - error TS2532: Object is possibly 'undefined'.

77         const customUpload = this.config.uploader.uploadByFile(files[0]);
                                ~~~~~~~~~~~~~~~~~~~~
src/uploader.ts:77:30 - error TS2532: Object is possibly 'undefined'.

77         const customUpload = this.config.uploader.uploadByFile(files[0]);
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/uploader.ts:77:30 - error TS2722: Cannot invoke an object which is possibly 'undefined'.

77         const customUpload = this.config.uploader.uploadByFile(files[0]);
```

Reply:
type guard is not working because of some problem. It's better to find it and resolve